x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/alibabacloudsearch/action/AlibabaCloudSearchCompletionAction.java

```diff
@@ -61,16 +59,14 @@ public void execute(InferenceInputs inferenceInputs, TimeValue timeout, ActionLi
             return;
         }
 
+        ActionListener<InferenceServiceResults> wrappedListener = wrapFailuresInElasticsearchException(
```

Comment:
These Alibaba classes are the outlier everything else is done by `SenderExecutableAction`. Please can you convert  `AlibabaCloudSearchActionCreator` to produce `SenderExecutableAction` and remove these classes. You will probably have to keep `AlibabaCloudSearchCompletionAction` as that has extra logic in it to check the inputs.